### PR TITLE
Prepare Fission for IPv6 uses

### PR DESCRIPTION
**Pull request by valentin2105**
Hi,

Here is a small contribution to allow Fission to talk to IPv6 function pods. 

I'm able to make all work except to communications to IPv6 function pod : 

poolmgr logs : 
```
 Error: 500: Post http://1404:f600:f::0:e:16:d0bb:8000/: invalid URL port "404:f600:f::0:e:16:d0bb"
```

---
This change is [Reviewable](https://reviewable.io/reviews/fission/fission/408)

### CODE REVIEW

File: executor/poolmgr/gp.go [hunk 1]
```diff
@@ -297,15 +297,28 @@ func (gp *GenericPool) getFetcherUrl(podIP string) string {
...
+}
+
 func (gp *GenericPool) getSpecializeUrl(podIP string, version int) string {
 	u := os.Getenv("TEST_SPECIALIZE_URL")
 	if len(u) != 0 {
 		return u
 	}
-	if version == 1 {
-		return fmt.Sprintf("http://%v:8888/specialize", podIP)
+	t := IsIPv6(podIP)
```
Comment (soamvasani) [inline executor/poolmgr/gp.go]: Thanks, @valentin2105 !

May I suggest reorganizing this if-else to set the host part of the URL separately from the version check? That way you won't need the nested if-s.

**Comment by valentin2105** [comment]
@soamvasani Thanks for the review. 

It seem better for you ?

File: executor/poolmgr/gp.go [hunk 2]
```diff
@@ -297,15 +297,28 @@ func (gp *GenericPool) getFetcherUrl(podIP string) string {
...
+}
+
 func (gp *GenericPool) getSpecializeUrl(podIP string, version int) string {
 	u := os.Getenv("TEST_SPECIALIZE_URL")
 	if len(u) != 0 {
 		return u
 	}
+	t := IsIPv6(podIP)
+	if t == false {
+		r := fmt.Sprintf("http://%v:8888", podIP)
```
Comment (soamvasani) [inline executor/poolmgr/gp.go]: This defines `r` only inside this block, so line 318 below causes a build failure. I think you'll have to define `r` as a var outside the if statement. 

(Also, what does `r` stand for? Maybe use just `baseUrl` or something? And could you also rename `t` to something like `v6`?)

**Comment by valentin2105** [comment]
@soamvasani It should be better ;)

File: executor/poolmgr/gp.go [hunk 3]
```diff
@@ -297,15 +297,29 @@ func (gp *GenericPool) getFetcherUrl(podIP string) string {
...
+
 func (gp *GenericPool) getSpecializeUrl(podIP string, version int) string {
 	u := os.Getenv("TEST_SPECIALIZE_URL")
+	isv6 := IsIPv6(podIP)
+	var baseUrl string
 	if len(u) != 0 {
 		return u
 	}
+	if isv6 == false {
+		baseUrl := fmt.Sprintf("http://%v:8888", podIP)
```
Comment (soamvasani) [inline executor/poolmgr/gp.go]: subtle thing, but this should be `baseUrl = ...` not `baseUrl := ...`. With the `:=` operator you're creating a new variable inside this scope and hiding the baseUrl variable defined on line 308.

**Comment by valentin2105** [comment]
I have made the correction, but the tests fail again.

**Comment by valentin2105** [comment]
@soamvasani  It seem ok for you ?

File: executor/poolmgr/gp.go [hunk 4]
```diff
@@ -297,15 +297,29 @@ func (gp *GenericPool) getFetcherUrl(podIP string) string {
 	return fmt.Sprintf("http://%v:8000/", podIP)
```
Comment (soamvasani) [inline executor/poolmgr/gp.go]: This needs the same change?

**Comment by valentin2105** [comment]
@soamvasani It seem good 😄  I always forget to FMT 🤣 .

**Comment by soamvasani** [comment]
👍 thanks @valentin2105 !

**Pull request merged**